packages/firebase_auth/firebase_auth_platform_interface/lib/src/platform_interface/platform_interface_user.dart

```diff
@@ -239,6 +239,102 @@ abstract class UserPlatform extends PlatformInterface {
     throw UnimplementedError('linkWithProvider() is not implemented');
   }
 
+  /// Renews the user’s authentication using the provided auth provider instance.
+  /// On Web you should use [linkWithPopup] instead.
+  ///
+  /// A [FirebaseAuthException] maybe thrown with the following error code:
+  /// - **provider-already-linked**:
```

Comment:
Not sure these codes are correct for this operation? Copied from somewhere else?

These don't belong here afaik;

- **provider-already-linked**
- **credential-already-in-use**
- **email-already-in-use**


(same again for popup api below this one)

Reply:
Fixed :) 
Also removed the phone errors since it would use reauthenticateWithCredentials